File: POLIMORFISMO/EJERCICIO-1/Videojuego.java
package poliformismo;

class Videojuego {
	private String nombre;
	private String plataforma;
	private int cantidadJugadores;
	
	public Videojuego() {
		this.nombre = "Desconocido";
		this.plataforma = "Desconocida";
		this.cantidadJugadores =1;
	}
	
	public Videojuego(String nombre, String plataforma) {
		this.nombre = nombre;
		this.plataforma = plataforma;
		this.cantidadJugadores = 1;
	}
	
	public Videojuego(String nombre, String plataforma, int cantidadJugadores) {
		this.nombre = nombre;
		this.plataforma = plataforma;
		this.cantidadJugadores = cantidadJugadores;
		
	}
	
	public void mostrar() {
		System.out.println("Nombre: " + nombre);
		System.out.println("Plataforma: " + plataforma);
		System.out.println("Cantidad de jugadores: " + cantidadJugadores);
	}
	
	public void agregarJugadores() {
		this.cantidadJugadores += 1;
		
	}
	public void agregarJugadores(int cantidad) {
		this.cantidadJugadores += cantidad;
		System.out.println("se agregaron " + cantidad);
	}
	
	public static void main(String[] args) {
		Videojuego jue1 = new Videojuego("mine", "pc");
		Videojuego jue2 = new Videojuego("fife", "palr", 5);
		
		jue1.mostrar();
        System.out.println();
        jue2.mostrar();
        
        jue1.agregarJugadores(); 
        jue2.agregarJugadores(3); 
        System.out.println();
        jue1.mostrar();
        System.out.println();
        jue2.mostrar();
	}
	
}

File: POLIMORFISMO/EJERCICIO-1/Videojuego.py
class Videojuego:
    def __init__(self, nombre="Desconocido", plataforma="Desconocida", cantidadJugadores= 1):
        self.__nombre = nombre
        self.__plataforma = plataforma
        self.__cantidadJugadores = cantidadJugadores
    def mostrar(self):
        print(f"nombre: {self.__nombre}")
        print(f"plataforma: {self.__plataforma}")
        print(f"Cantidad de jugadores: {self.__cantidadJugadores}")

    def agregarJugadores(self, cantidad = 1):
        self.__cantidadJugadores += cantidad
        print(f"se agregaron {cantidad}")
    
jug1 = Videojuego()
jug2 = Videojuego("pepe", "Steam")
jug3 = Videojuego("due", "Play", 10)

jug1.mostrar()
print()
jug2.mostrar()
print()
jug3.mostrar()

jug1.agregarJugadores()  
jug2.agregarJugadores(3) 

print("\nDespués de agregar jugadores:")
jug1.mostrar()
print()
jug2.mostrar()